core/src/main/java/de/flo56958/MineTinker/Modifiers/Types/Smite.java

```diff
@@ -0,0 +1,154 @@
+                Main.getPlugin());
+    }
+
+    @Override
+    public List<Enchantment> getAppliedEnchantments() {
+        List<Enchantment> enchantments = new ArrayList<>();
+        enchantments.add(Enchantment.DAMAGE_UNDEAD);
+
+        return enchantments;
+    }
+
+    @Override
+    public void reload() {
+        FileConfiguration config = getConfig();
+        config.options().copyDefaults(true);
+
+        String key = "Smite";
+        config.addDefault(key + ".allowed", true);
+        config.addDefault(key + ".name", key);
+        config.addDefault(key + ".name_modifier", "Holy Bone");
+        config.addDefault(key + ".modifier_item", "BONE"); //Needs to be a viable Material-Type
+        config.addDefault(key + ".description", "Weapon does additional damage towards the Undead!");
+        config.addDefault(key + ".description_modifier", "%YELLOW%Modifier-Item for the Smite-Modifier");
+        config.addDefault(key + ".Color", "%YELLOW%");
+        config.addDefault(key + ".MaxLevel", 5);
+        config.addDefault(key + ".CompatibleWithSharpness", false);
+        config.addDefault(key + ".CompatibleWithArthropods", false);
+        config.addDefault(key + ".Recipe.Enabled", true);
+        config.addDefault(key + ".Recipe.Top", "BMB");
+        config.addDefault(key + ".Recipe.Middle", "MIM");
+        config.addDefault(key + ".Recipe.Bottom", "BMB");
+        config.addDefault(key + ".Recipe.Materials.B", "BONE");
+        config.addDefault(key + ".Recipe.Materials.M", "BONE_MEAL");
+        config.addDefault(key + ".Recipe.Materials.I", "IRON_INGOT");
+
+        ConfigurationManager.saveConfig(config);
+
+        init(config.getString(key + ".name"),
+                "[" + config.getString(key + ".name_modifier") + "] " + config.getString(key + ".description"),
+                ChatWriter.getColor(config.getString(key + ".Color")), config.getInt(key + ".MaxLevel"),
+                modManager.createModifierItem(Material.getMaterial(config.getString(key + ".modifier_item")),
+                ChatWriter.getColor(config.getString(key + ".Color")) + config.getString(key + ".name_modifier"),
+                ChatWriter.addColors(config.getString(key + ".description_modifier")), this));
+
+        this.compatibleWithSharpness = config.getBoolean(key + ".CompatibleWithSharpness");
+        this.compatibleWithArthropods = config.getBoolean(key + ".CompatibleWithArthropods");
+    }
+
+    @Override
+    public ItemStack applyMod(Player p, ItemStack tool, boolean isCommand) {
+        if (Modifier.checkAndAdd(p, tool, this, "smite", isCommand) == null) {
+            return null;
+        }
+
+        ItemMeta meta = tool.getItemMeta();
+
+        if (meta != null) {
+            if (!ToolType.AXE.getMaterials().contains(tool.getType()) && !ToolType.SWORD.getMaterials().contains(tool.getType())) {
+                if (!this.compatibleWithSharpness) {
+                    if (modManager.get(ModifierType.SHARPNESS) != null) {
```

Comment:
A better way to do this is `Sharpness.instance()`. Every Modifier is always not null (except in the `get`-Method). So you can just call `modManager.hasMod(Sharpness.instance())`. `modManager.get(ModifierType.SHARPNESS) != null` is from the older versions of the ModManager (before NBT was added) now the ModManager just checks the NBT-tag from the `ModifierType`-Enum. Actually the `get()`- and the `getAdmin()`-Methods are no longer necessary.

Makes a cleaner code.

Reply:
I believe I copied this code almost exactly from the ender modifier, if I modify this I'll likely modify ender to meet the same standard.